extensions/InputBox/InputBox.hooks.php

```diff
@@ -52,24 +52,38 @@ public static function onMediaWikiPerformAction(
 		$request, 
 		$wiki )
 	{
+		global $wgScript, $wgUser;
+
 		if( $wiki->getAction( $request ) !== 'edit' ){
 			# not our problem
 			return true;
 		}
+
+		$params = $request->getValues();
+
+		if ( $request->getText( 'preload', '' ) !== '' ){
+			# Make sure any call with preload uses VE when it matches the user's prefs
+			if ( EditorPreference::isVisualEditorPrimary() && $wgUser->isLoggedIn() ) {
+				$params['veaction'] = 'edit';
+				unset( $params['action'] );
+
+				$output->redirect( wfAppendQuery( $wgScript, $params ), '301' );
+				return false;
```

Comment:
In our use case, we only use preload.

However, there is nothing preventing somebody for adding both preload and prefix, so I should probably add in a check for that.

How about the following at line 70:

`		if( $request->getText( 'prefix', '' ) !== '' && isset( $params['title'] ) ) {
			$title = $params['prefix'];
			$title .= $params['title'];
			$params['title'] = $title;
			unset( $params['prefix'] );
		}
`